kit/service/transport.go

```diff
@@ -141,6 +141,11 @@ func (r *Router) MountRoute(route Route) *Router {
 		_, handler = newrelic.WrapHandle(r.newrelicApp, route.GetPath(), handler)
 	}
 
+	if route.CORSEnabled() {
+		handler = corsMiddleware(handler)
+		r.prefixMux.Methods("OPTIONS").Path(route.GetPath()).Handler(handler)
```

Comment:
This shouldn't wrap the request handler, you should provide a dummy handler that always responds 200 OK. Otherwise it'll try to process the actual request when receiving the pre-flight.